### PR TITLE
Speed up by caching aws resources and clients

### PR DESCRIPTION
**Pull request by l2ol33rt**
I'd like to propose this change which speeds up credstash. This improve was achieved by caching the the `kms` client and `dynamodb` resource instead of getting a new one each time during `credstash getall`.

As an example I've got 200+ rows in my dynamodb and here are the results `credstash getall` when compared against `credstash 1.12.0`:

`credstash 1.12.0`:

```
$ credstash list | wc -l                                                    
218
$ time credstash getall
...
real    0m56.437s
user    0m4.605s
sys     0m0.191s
```

This PR's version of credstash:

```
$ credstash list | wc -l
218
$ time credstash getall
real    0m14.232s
user    0m0.978s
sys     0m0.051s
```

Let me know what you think!

~Rob

### CODE REVIEW

**Comment by l2ol33rt** [comment]
@alex-luminal any thoughts?

File: credstash.py [hunk 1]
```diff
@@ -259,19 +259,28 @@ def putSecret(name, secret, version="", kms_key="alias/credstash",
...
+                  context=None, session=None, **kwargs):
     '''
     fetch and decrypt all secrets
     '''
     output = {}
     secrets = listSecrets(region, table, **kwargs)
+
+    # We only need these things once
+    if session is None:
+        session = get_session(**kwargs)
```
Comment (toidi) [inline credstash.py]: `get_session` has caching mechanism
https://github.com/l2ol33rt/credstash/blob/master/credstash.py#L366-L374

No need to repeat it.

**Comment by l2ol33rt** [comment]
@alex-luminal no problem, Ill take a look at whats broken this week and get back to you.

**Comment by alex-luminal** [comment]
Awesome!

**Comment by alex-luminal** [comment]
👍

**Pull request merged**